torch/linalg/__init__.py

```diff
@@ -140,6 +140,110 @@
     (tensor(3.7417), tensor(11.2250))
 """)
 
+eigh = _add_docstr(_linalg.linalg_eigh, r"""
+linalg.eigh(input, UPLO='L') -> tuple(Tensor, Tensor)
+
+This function returns eigenvalues and eigenvectors
+of a complex Hermitian (conjugate symmetric) or real symmetric matrix :attr:`input`
+represented by a namedtuple (eigenvalues, eigenvectors).
+
+This function calculates all eigenvalues (and vectors) of :attr:`input`
+such that :math:`\text{input} = V \text{diag}(e) V^H`, where :math:`e` is eigenvalues tensor
+and :math:`V` is eigenvectors tensor.
+
+Since the input matrix :attr:`input` is supposed to be Hermitian,
+only the lower triangular portion is used by default
+and the imaginary part of the diagonal will always be treated as zero.
+
+.. note:: The eigenvalues of real symmetric or complex Hermitian matrices are always real.
+
+.. note:: The eigenvalues/eigenvectors are computed using LAPACK/MAGMA routines ``_syevd``, ``_heevd``.
+
+Args:
+    input (Tensor): the input tensor of size :math:`(*, n, n)` consisting of Hermitian
+                    :math:`n \times n` matrices, where `*` is zero or more batch dimensions.
+    UPLO ('L', 'U', optional): controls whether to consider upper-triangular or lower-triangular part.
+        Default: ``'L'``
+
+Returns:
+    (Tensor, Tensor): A namedtuple (eigenvalues, eigenvectors) containing
+
+        - **eigenvalues** (*Tensor*): Shape :math:`(*, m)`.
+            The eigenvalues in ascending order, each repeated according to its multiplicity.
```

Comment:
Is the phrase "each repeated according to its multiplicity" necessary?

Reply:
I guess not